core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -176,6 +181,7 @@ private Builder(ViewMetadata base) {
       this.currentVersionId = base.currentVersionId();
       this.location = base.location();
       this.uuid = base.uuid();
+      this.metadataLocation = null;
```

Comment:
I just want to note that we're deviating from `TableMetadata` behavior a little in this builder, since it relates to the `metadataLocation`. In `TableMetadata.Builder`, changes are accumulated across builder instances. Basically the changes get reset to the empty set when the new table metadata is written out to an object store and reloaded from that copy. In this builder, we accumulate a new set of changes in every builder and don't carry them through. The difference is that the table metadata needs to accumulate changes across multiple operations in a transaction, while but there are no transactions for view metadata because every changes affects only the view metadata object (not other data or metadata files).